<launch>
    <!-- ORB-SLAM3 -->
    <node name="orb_slam3_stereo_acc" pkg="orb_slam3_ros_wrapper" type="orb_slam3_ros_wrapper_stereo_acc" output="screen">
        <!-- for TUM dataset -->
        <remap from="/camera/left/image_raw"    to="/cam0/image_raw"/>
        <remap from="/camera/right/image_raw"   to="/cam1/image_raw"/>
        <remap from="/camera/left/keypoints"   to="/cam0/keypoints"/>
        <remap from="/camera/right/keypoints"   to="/cam1/keypoints"/>
        <remap from="/camera/left/descriptors"   to="/cam0/descriptors"/>
        <remap from="/camera/right/descriptors"   to="/cam1/descriptors"/>

        <!-- Parameters for original ORB-SLAM3 -->
        <param name="voc_file"      type="string"   value="$(find orb_slam3_ros_wrapper)/config/ORBvoc.txt" />
        <param name="settings_file" type="string"   value="$(find orb_slam3_ros_wrapper)/config/TUM-VI.yaml" />

        <!-- Parameters for ROS -->
        <param name="world_frame_id"    type="string"   value="world" />
        <param name="cam_frame_id"      type="string"   value="camera" />
    </node>

    <!-- Visualization - RViz-->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find orb_slam3_ros_wrapper)/config/orb_slam3_no_imu.rviz" output="screen" />

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_orb_slam3" output="screen" ns="orb_slam3_ros" >
        <param name="/target_frame_name" value="/world" />
        <param name="/source_frame_name" value="/camera" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node>
</launch>